backend/apis/clients/integrations/patch-integration.api.js

```diff
@@ -0,0 +1,97 @@
+const { app, pool, invalidRequest, databaseError } = require("../../../server");
+const {
+  checkValid,
+  validId,
+  validEnum,
+  nullableValidEnum,
+  nullableNonEmptyString
+} = require("../../utils/validation-utils");
+const { getClientById } = require("../get-client.api");
+const {
+  integrationTypes,
+  getIntegrationName
+} = require("./integrations-utils");
+const mysql = require("mysql");
+
+app.patch("/api/clients/:clientId/integrations/:integrationId", (req, res) => {
+  const validationErrors = [
+    ...checkValid(
+      req.params,
+      validId("clientId"),
+      validEnum("integrationId", ...Object.keys(integrationTypes))
+    ),
+    ...checkValid(
+      req.body,
+      nullableNonEmptyString("externalId"),
+      nullableValidEnum("status", "disabled", "enabled", "broken")
+    )
+  ];
+
+  if (validationErrors.length > 0) {
+    return invalidRequest(res, validationErrors);
+  }
+
+  const clientId = Number(req.params.clientId);
+  const integrationId = req.params.integrationId;
+
+  getClientById(clientId, (err, client) => {
+    if (err) {
+      return databaseError(rewq, res, err);
```

Comment:
req

Reply:
👍 good catch. I have updated the code to fix this. Please re-review.